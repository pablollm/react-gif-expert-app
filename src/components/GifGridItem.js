import React from 'react'

export const GifGridItem = ({gif}) => {
  return (
        <div className="card">
            <img src={gif.url} alt={gif.title}/>
            <p className='card-title'>{gif.title}</p>
        </div>
  )
}
