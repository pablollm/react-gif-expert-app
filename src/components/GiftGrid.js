import React from 'react'
import { useEffect ,useState} from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../services/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GiftGrid = ({category}) => {
const {data:gifs,loading} = useFetchGifs(category);
    
   
  return (
    <>
    <h3>{category}</h3>
    {loading && <p>Loading...</p>}
     <div className='card-grid'>
      {
        gifs.map(gif => (  <GifGridItem key={gif.id} gif={gif}/>))
     }
     </div>

    </>
  )
}

GiftGrid.propTypes = {
category: PropTypes.string.isRequired
}