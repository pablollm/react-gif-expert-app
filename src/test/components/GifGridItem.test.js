import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {
   const title   = 'Title';
   const url  = 'https://media1.giphy.com/media/l0HlJn1y7z2QkQ7p2/200w.gif';
   const wrapper = shallow(<GifGridItem  title={title} url={url} />);

     test('debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
     });

     test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
     });

     test('debe tener una imagen con el url', () => {
         const img = wrapper.find('img');
         expect(img.prop('src')).toBe(url);
     })
})