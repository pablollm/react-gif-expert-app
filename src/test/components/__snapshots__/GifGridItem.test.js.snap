// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Title"
    src="https://media1.giphy.com/media/l0HlJn1y7z2QkQ7p2/200w.gif"
  />
  <p
    className="card-title"
  >
    Title
  </p>
</div>
`;
